sky/provision/paperspace/instance.py

```diff
@@ -0,0 +1,313 @@
+    for inst_id, inst in instances.items():
+        logger.debug(f'Terminating instance {inst_id}')
+        if worker_only and inst['name'].endswith('-head'):
+            continue
+        try:
+            client.remove(inst_id)
+        except Exception as e:  # pylint: disable=broad-except
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError(
+                    f'Failed to terminate instance {inst_id}: '
+                    f'{common_utils.format_exception(e, use_bracket=False)}'
+                ) from e
+
+    # TODO(asaiacai): Possible private network resource leakage for autodown
+    if not worker_only:
+        try:
+            time.sleep(POLL_INTERVAL)
+            network = client.get_network(network_name=cluster_name_on_cloud)
+            client.delete_network(network_id=network['id'])
+        except IndexError:
+            logger.warning(f'Network {cluster_name_on_cloud}'
+                           'already deleted')
+
+
+def get_cluster_info(
+    region: str,
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> common.ClusterInfo:
+    del region, provider_config  # unused
+    running_instances = _filter_instances(cluster_name_on_cloud, ['ready'])
+    instances: Dict[str, List[common.InstanceInfo]] = {}
+    head_instance_id = None
+    for instance_id, instance_info in running_instances.items():
+        instances[instance_id] = [
+            common.InstanceInfo(
+                instance_id=instance_id,
+                internal_ip=instance_info['privateIp'],
+                external_ip=instance_info['publicIp'],
+                ssh_port=22,
+                tags={},
+            )
+        ]
+        if instance_info['name'].endswith('-head'):
+            head_instance_id = instance_id
+
+    return common.ClusterInfo(
+        instances=instances,
+        head_instance_id=head_instance_id,
+    )
+
+
+def query_instances(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+    non_terminated_only: bool = True,
+) -> Dict[str, Optional[status_lib.ClusterStatus]]:
+    """See sky/provision/__init__.py"""
+    assert provider_config is not None, (cluster_name_on_cloud, provider_config)
+    instances = _filter_instances(cluster_name_on_cloud, None)
+
+    status_map = {
+        'starting': status_lib.ClusterStatus.INIT,
+        'restarting': status_lib.ClusterStatus.INIT,
+        'upgrading': status_lib.ClusterStatus.INIT,
+        'provisioning': status_lib.ClusterStatus.INIT,
+        'stopping': status_lib.ClusterStatus.INIT,
+        'serviceready': status_lib.ClusterStatus.INIT,
+        'ready': status_lib.ClusterStatus.UP,
+        'off': status_lib.ClusterStatus.STOPPED,
+    }
+    statuses: Dict[str, Optional[status_lib.ClusterStatus]] = {}
+    for inst_id, inst in instances.items():
+        status = status_map[inst['state']]
+        if non_terminated_only:
+            continue
```

Comment:
Oops, this line seems causing the `sky status -r` removing all clusters from the table. Does Paperspace has any state that indicate a VM is terminated? If not, we should probably remove these two lines.
```suggestion
```